diff --git a/node_modules/@hethers/wallet/lib.esm/index.js b/node_modules/@hethers/wallet/lib.esm/index.js
index 6bf7665..fde9acd 100644
--- a/node_modules/@hethers/wallet/lib.esm/index.js
+++ b/node_modules/@hethers/wallet/lib.esm/index.js
@@ -158,9 +158,13 @@ export class Wallet extends Signer {
         }));
     }
     signMessage(message) {
-        return __awaiter(this, void 0, void 0, function* () {
+        if (typeof unityEnv != 'undefined') {
             return joinSignature(this._signingKey().signDigest(hashMessage(message)));
-        });
+        } else {
+            return __awaiter(this, void 0, void 0, function* () {
+                return joinSignature(this._signingKey().signDigest(hashMessage(message)));
+            });
+        }
     }
     _signTypedData(domain, types, value) {
         return __awaiter(this, void 0, void 0, function* () {
